R/nodeCentricSteinerForestProblem-class.R

```diff
@@ -160,7 +163,9 @@ nodeCentricSteinerForestProblem <- R6Class("nodeCentricSteinerForestProblem",
       while(length(private$fixedTerminalIndices) == 0){
         
         #Looks complicated, but really it just resamples the isTerminal/fixedTerminal nodeIDs
-        private$fixedTerminalIndices <- super$getNodeDT()[isTerminal == TRUE, .SD[sample(c(TRUE,FALSE), length(isTerminal), replace = TRUE, prob = c(pSuccess,1-pSuccess)), .nodeID]]
+        private$fixedTerminalIndices <- super$getNodeDT()[isTerminal == TRUE, 
+                                                          .SD[sample(c(TRUE,FALSE), length(isTerminal), replace = TRUE, prob = c(pSuccess,1-pSuccess)), .nodeID]
+                                                          ] %>% unique()# make sure the fixedTerminal are unique
```

Comment:
sampling **with replacement** could lead to duplicated terminal which I think could be the reason why I had really large networks when I did 1000 bootstraps. I can't be 100% sure though.

Reply:
Hmmm. I don't follow the logic - we use sets in the backend, so duplicated values are removed.

Any chance that you could come up with a minimum reproducible example? Obviously it cannot contain any confidential data, so it would need to use public data.